File: test57_out/test57/ClsFuncRef/ClsFuncRef.js

/*
 * ======== CLASS test57.ClsFuncRef ========
 */

var $$c = function() {
    const ClsFuncRef = {};
    ClsFuncRef.ClsFuncRef = ClsFuncRef
    ClsFuncRef.$name = 'test57.ClsFuncRef';
    ClsFuncRef.pollen$used = 0;
    ClsFuncRef.ClsFuncRef$$id = 1;
    ClsFuncRef.ClsFuncRef = function(cn) {
        this.$$id = ClsFuncRef.ClsFuncRef$$id++;
        this.$$cname = cn ? cn : 'test57_ClsFuncRef_ClsFuncRef__'+this.$$id+'__S';
        this.$$tname = 'test57_ClsFuncRef_ClsFuncRef';
        this.$$uname = 'test57.ClsFuncRef';
        this.myHandler1_Host = 0;
        this.myHandler2_Host = 0;
        this.myHandler3_Host = 0;
        this.myHandler4_Host = 0;
        this.flag = true;
        this.myHandler1_Targ = 0;
        this.myHandler2_Targ = 0;
        this.myHandler3_Targ = 0;
        this.myHandler4_Targ = 0;
        this.flag2 = true;
        this.tt = 0;
    }
    ClsFuncRef.ClsFuncRef.prototype = new $$Struct('test57.ClsFuncRef.ClsFuncRef');
    ClsFuncRef.ClsFuncRef.prototype.$$isAggFld = {};
    ClsFuncRef.ClsFuncRef.prototype.$$isAggFld.myHandler1_Host = false;
    ClsFuncRef.ClsFuncRef.prototype.$$isAggFld.myHandler2_Host = false;
    ClsFuncRef.ClsFuncRef.prototype.$$isAggFld.myHandler3_Host = false;
    ClsFuncRef.ClsFuncRef.prototype.$$isAggFld.myHandler4_Host = false;
    ClsFuncRef.ClsFuncRef.prototype.$$isAggFld.flag = false;
    ClsFuncRef.ClsFuncRef.prototype.$$isAggFld.myHandler1_Targ = false;
    ClsFuncRef.ClsFuncRef.prototype.$$isAggFld.myHandler2_Targ = false;
    ClsFuncRef.ClsFuncRef.prototype.$$isAggFld.myHandler3_Targ = false;
    ClsFuncRef.ClsFuncRef.prototype.$$isAggFld.myHandler4_Targ = false;
    ClsFuncRef.ClsFuncRef.prototype.$$isAggFld.flag2 = false;
    ClsFuncRef.ClsFuncRef.prototype.$$isAggFld.tt = false;
    ClsFuncRef.ClsFuncRef.prototype.new_host = function() {
        var $$text = '';
        this.$$privateInit();
        this.setTheFuncRefOnHost1((ClsFuncRef.hostTheHandler));
        this.setTheFuncRefOnHost2((ClsFuncRef.targetTheHandler));
        this.setTheFuncRefOnHost3((ClsFuncRef.hostTheHandler));
        this.setTheFuncRefOnHost4((ClsFuncRef.targetTheHandler));
        this.$$hostInitLastPass();
        return this;
    }
    ClsFuncRef.ClsFuncRef.prototype.setTheFuncRefOnHost1 = function( h ) {
        var $$text = '';
        (this.myHandler1_Host) = (h);
        this.$$hostInitLastPass();
        return $$text;
    }
    ClsFuncRef.ClsFuncRef.prototype.setTheFuncRefOnHost2 = function( h ) {
        var $$text = '';
        (this.myHandler2_Host) = (h);
        this.$$hostInitLastPass();
        return $$text;
    }
    ClsFuncRef.ClsFuncRef.prototype.setTheFuncRefOnHost3 = function( h ) {
        var $$text = '';
        (this.myHandler3_Targ) = (h);
        this.$$hostInitLastPass();
        return $$text;
    }
    ClsFuncRef.ClsFuncRef.prototype.setTheFuncRefOnHost4 = function( h ) {
        var $$text = '';
        (this.myHandler4_Targ) = (h);
        this.$$hostInitLastPass();
        return $$text;
    }
    ClsFuncRef.ClsFuncRef.prototype.hostTheHandler = function() {
        var $$text = '';
         printf("host Handler called\n") ;

        this.$$hostInitLastPass();
        return $$text;
    }
    ClsFuncRef.pollen__uses__ = function() {
    }
    ClsFuncRef.ClsFuncRef.prototype.$$privateInit = function() {
        this.flag = true;
        this.flag2 = true;
        ClsFuncRef.tt = new $$Ref('test57_ClsFuncRef_ClsFuncRef_foo__E');
    }
    ClsFuncRef.ClsFuncRef.prototype.$$hostInitLastPass = function() {
    }
    return ClsFuncRef;
}
var $$u = $$c();
$units['test57.ClsFuncRef'] = $$u;
$units.push($$u);

File: test57_out/test57/ClsFuncRef/ClsFuncRef.c

/*
 * ======== string literals ========
 */

static const char test57_ClsFuncRef__s__fileName[] = "ClsFuncRef.p";
static const char test57_ClsFuncRef__s__0[] = "ClsFuncRef";

/*
 * ======== code injections ========
 */

 #include <stdio.h> 

/*
 * ======== function definitions ========
 */

void test57_ClsFuncRef_ClsFuncRef_setTheFuncRefOnTarg1__I( test57_ClsFuncRef_ this, test57_ClsFuncRef_HandlerP_handler h ) {
    (this->myHandler3_Host) = (h);
}

void test57_ClsFuncRef_ClsFuncRef_setTheFuncRefOnTarg2__I( test57_ClsFuncRef_ this, test57_ClsFuncRef_HandlerP_handler h ) {
    (this->myHandler4_Host) = (h);
}

void test57_ClsFuncRef_ClsFuncRef_setTheFuncRefOnTarg3__I( test57_ClsFuncRef_ this, test57_ClsFuncRef_HandlerP_handler h ) {
    (this->myHandler1_Targ) = (h);
}

void test57_ClsFuncRef_ClsFuncRef_setTheFuncRefOnTarg4__I( test57_ClsFuncRef_ this, test57_ClsFuncRef_HandlerP_handler h ) {
    (this->myHandler2_Targ) = (h);
}

void test57_ClsFuncRef_ClsFuncRef_targetTheHandler__I( test57_ClsFuncRef_ this ) {
     printf("target Handler called\n") ;

}

void test57_ClsFuncRef_ClsFuncRef_foo__E( test57_ClsFuncRef_ this ) {
    test57_ClsFuncRef_ClsFuncRef_setup__I(this);
     printf("check myHandler1_Host\n") ;

    if ((this->myHandler1_Host) != null) {
        this->myHandler1_Host();
    }
     printf("check myHandler2_Host\n") ;

    if ((this->myHandler2_Host) != null) {
        this->myHandler2_Host();
    }
     printf("check myHandler3_Host\n") ;

    if ((this->myHandler3_Host) != null) {
        this->myHandler3_Host();
    }
     printf("check myHandler4_Host\n") ;

    if ((this->myHandler4_Host) != null) {
        this->myHandler4_Host();
    }
     printf("check myHandler1_Targ\n") ;

    if ((this->myHandler1_Targ) != null) {
        this->myHandler1_Targ();
    }
     printf("check myHandler2_Targ\n") ;

    if ((this->myHandler2_Targ) != null) {
        this->myHandler2_Targ();
    }
     printf("check myHandler3_Targ\n") ;

    if ((this->myHandler3_Targ) != null) {
        this->myHandler3_Targ();
    }
     printf("check myHandler4_Targ\n") ;

    if ((this->myHandler4_Targ) != null) {
        this->myHandler4_Targ();
    }
}

test57_ClsFuncRef_ test57_ClsFuncRef_ClsFuncRef_new___E( test57_ClsFuncRef_ this ) {
    return this;
}

void test57_ClsFuncRef_ClsFuncRef_setup__I( test57_ClsFuncRef_ this ) {
    test57_ClsFuncRef_ClsFuncRef_setTheFuncRefOnTarg2__I(this, (test57_ClsFuncRef_ClsFuncRef_targetTheHandler__I));
    test57_ClsFuncRef_ClsFuncRef_setTheFuncRefOnTarg4__I(this, (test57_ClsFuncRef_ClsFuncRef_targetTheHandler__I));
}

